api_tests/tests/sanity.ts

```diff
@@ -152,4 +160,31 @@ describe("Sanity", () => {
             ).rejects.toMatchObject(expectedProblem);
         })
     );
+
+    it(
+        "bob-connects-to-alice-using-config",
+        createAliceAndBob(async ([alice, bob]) => {
+            await alice.cndInstance.start();
+
+            const aliceId = await alice.cnd.getPeerId();
+
+            const aliceAddresses = await alice.cnd.getPeerListenAddresses();
+
+            const configOverride = {
+                network: {
+                    peer_addresses: aliceAddresses,
+                },
+            };
+
+            const currentConfig = bob.cndInstance.getConfigFile();
+
+            const updatedConfig = merge(configOverride, currentConfig);
+
+            bob.cndInstance.setConfigFile(updatedConfig);
+
+            await bob.cndInstance.start();
+
+            await bob.pollUntilConnectedTo(aliceId);
```

Comment:
Can remove some newlines to group related functionality :)

```suggestion
            await alice.cndInstance.start();

            const aliceAddresses = await alice.cnd.getPeerListenAddresses();
            const configOverride = {
                network: {
                    peer_addresses: aliceAddresses,
                },
            };

            const currentConfig = bob.cndInstance.getConfigFile();
            const updatedConfig = merge(configOverride, currentConfig);

            bob.cndInstance.setConfigFile(updatedConfig);
            await bob.cndInstance.start();

            const aliceId = await alice.cnd.getPeerId();
            await bob.pollUntilConnectedTo(aliceId);
```